### PR TITLE
Fix:#4197: validate post title must be shorter than 200 letters

### PR DESCRIPTION
**Pull request by varshard**
https://github.com/LemmyNet/lemmy/issues/4197

Fix post title validation to prevent creating a post title that's longer than 200 characters via the POST /post API.

### CODE REVIEW

**Review by dessalines — APPROVED** [review]

File: crates/utils/src/utils/validation.rs [hunk 1]
```diff
@@ -150,7 +147,8 @@ pub fn is_valid_matrix_id(matrix_id: &str) -> LemmyResult<()> {
 }
 
 pub fn is_valid_post_title(title: &str) -> LemmyResult<()> {
-  let check = VALID_POST_TITLE_REGEX.is_match(title) && !has_newline(title);
+  let length = title.trim().len();
+  let check = length >= 3 && length <= 200 && !has_newline(title);
```
Comment (dessalines) [inline crates/utils/src/utils/validation.rs:151]: You can also do this with [range contains](https://stackoverflow.com/questions/28265036/how-can-i-test-if-a-value-lies-within-a-range) but I leave that up to you.

Comment (dessalines) [reply crates/utils/src/utils/validation.rs:151]: Clippy beat me to it :laughing: 

https://woodpecker.join-lemmy.org/repos/129/pipeline/4064/17

Comment (varshard) [reply crates/utils/src/utils/validation.rs:151]: Thank you. I'm learning Rust by contributing and your feedback helps.

Comment (dessalines) [reply crates/utils/src/utils/validation.rs]: No probs!

**Review by dessalines — APPROVED** [review]

**Review by Nutomic — APPROVED** [review]

**Pull request merged**